repos/keg-components/configs/rollup.config.js

```diff
@@ -7,12 +7,142 @@ import cleanup from 'rollup-plugin-cleanup'
 import sourcemaps from 'rollup-plugin-sourcemaps'
 import alias from '@rollup/plugin-alias'
 import buildHook from './buildHook'
+
 const { getAliases } = require('./aliases.config')
 
 const { DEV_MODE, BUILD_HOOK } = process.env
 const babelConfig = require('./babel.config.js')
 
+const path = require('path')
+const fs = require('fs')
+
+/**
+ * gets the files at a given directory
+ * @param {string} dir 
+ * @param {Array<string>} exclude - used to filter out the files 
+ * 
+ * @returns {Array<string>} - a list of the file names with extension
+ */
+const getFilesFromDir = (dir, exclude=[]) => {
+  const files = fs.readdirSync(dir)
+  return exclude.length 
+    ? files.filter(file => !exclude.some(excludeItem => file.includes(excludeItem))) 
+    : files
+}
+
+const parentFolders = [
+  'components',
+  'hocs',
+  'hooks',
+  'utils'
+]
+
+const getIgnoreList = (isNative) => [
+  // folders
+  'internal',
+  '__tests__',
+
+  // files
+  '.stories.',
+  '.examples.',
+  '.wrapper.',
+  'index.js',
+  !isNative && '.native.'
+]
+
+/**
+ * returns the .native counterpart if it exists
+ * @param {string} current - current filename
+ * @param {Array<string>} list - array containing the files available
+ * 
+ * @returns {string|null}
+ */
+const getNativeFile = (current='', list=[]) => {
+  return list.find(element => 
+    element === current.replace('.js', '.native.js')
+  ) || null
+}
+
+/**
+ * Creates the mapping per directory
+ * @param {string} dir
+ * @param {boolean} isNative - if we're building for native or not
+ * @param {Array<string>} items - array containing filenames and/or folder names
+ * @param {Array<string>} ignoreList - which file/folders to ignore
+ * 
+ * @returns {object}
+ */
+const createExportMappings = (dir, isNative, items=[], ignoreList=[]) => {
+
+  const obj = {}
+  items.forEach((item) => {
```

Comment:
Nothing wrong here, but I typically do a `reduce` here
That way I don't have to define an external variable `obj`, and I can return on the same line.
Looks like this => 
```
  return items.reduce((obj, item) => {
    // ... all the codes

    return obj
  }, {})
```

Reply:
updated